source/AMDSClientUi.h

```diff
@@ -0,0 +1,99 @@
+#ifndef AMDSCLIENTUI_H
+#define AMDSCLIENTUI_H
+
+#include <QDialog>
+
+class QDialogButtonBox;
+class QLabel;
+class QLineEdit;
+class QPushButton;
+class QNetworkSession;
+class QDateTimeEdit;
+class QTextEdit;
+class QComboBox;
+class QListView;
+class QSpinBox;
+class QCheckBox;
+
+class AMDSClientAppController;
+class AMDSClientRequest;
+
+class AMDSClientUi : public QDialog
+{
+	Q_OBJECT
+public:
+	explicit AMDSClientUi(QWidget *parent = 0);
+	~AMDSClientUi();
+
+private slots:
+	/// connect to a server with the input hostName and portNumber
+	void connectToServer();
+
+	/// ============= SLOTs to handle AMDSClientAppController signals =========
+	/// slot to handle the signal of networkSessionOpening
+	void onNetworkSessionOpening();
+	/// slot to handle the signal of networkSessionOpened
+	void onNetworkSessionOpened();
+
+	/// slot to handle the signal of newServerConnected (add the serverIdentifier to the combox and update the ui displays -- buffernames and active connections)
+	void onNewServerConnected(QString serverIdentifier);
```

Comment:
All of the `QString` parameters should be `const QString &` and the `QStringList &` should be `const QStringList &`.